src/Controller/Admin/UserController.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Controller\Admin;
+
+use App\Entity\User;
+use App\Form\UserUpdateFormType;
+use App\Repository\UserRepository;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+class UserController extends AbstractController
+{
+    private UserRepository $userRepository;
+
+    private EntityManagerInterface $entityManager;
+
+    public function __construct(
+        UserRepository $userRepository,
+        EntityManagerInterface $entityManager
+    ) {
+        $this->userRepository = $userRepository;
+        $this->entityManager = $entityManager;
+    }
+
+    /**
+     * @Route(path="/users", methods={"GET"}, name="list_users")
+     */
+    public function listAllUsers(Request $request): Response
+    {
+        $query = $request->query->get('page');
```

Comment:
this variable should pe called `$page` and you could provide a default value for it (default page is 1). If no page is provided, you shouldn't list the entire db